src/main/java/com/lishid/openinv/ConfigUpdater.java

```diff
@@ -0,0 +1,161 @@
+    }
+
+    public void checkForUpdates() {
+        if (isConfigOutdated()) {
+            plugin.getLogger().info("[Config] Update found! Performing update...");
+            updateConfig();
+        } else {
+            plugin.getLogger().info("[Config] Update not found. Config is already up-to-date.");
+        }
+    }
+
+    private void updateConfig() {
+        // Get the old config settings
+        int itemOpenInvItemId = plugin.getConfig().getInt("ItemOpenInvItemID", 280);
+        boolean checkForUpdates = plugin.getConfig().getBoolean("CheckForUpdates", true);
+        boolean notifySilentChest = plugin.getConfig().getBoolean("NotifySilentChest", true);
+        boolean notifyAnyChest = plugin.getConfig().getBoolean("NotifyAnyChest", true);
+
+        Map<UUID, Boolean> anyChestToggles = null;
+        Map<UUID, Boolean> itemOpenInvToggles = null;
+        Map<UUID, Boolean> silentChestToggles = null;
+
+        if (plugin.getConfig().isSet("AnyChest")) {
+            anyChestToggles = updateAnyChestToggles();
+        }
+
+        if (plugin.getConfig().isSet("ItemOpenInv")) {
+            itemOpenInvToggles = updateItemOpenInvToggles();
+        }
+
+        if (plugin.getConfig().isSet("SilentChest")) {
+            silentChestToggles = updateSilentChestToggles();
+        }
+
+        // Clear the old config
+        for (String key : plugin.getConfig().getKeys(false)) {
+            plugin.getConfig().set(key, null);
+        }
+
+        // Set the new config options
+        plugin.getConfig().set("config-version", CONFIG_VERSION);
+        plugin.getConfig().set("check-for-updates", checkForUpdates);
+        plugin.getConfig().set("items.open-inv", getMaterialById(itemOpenInvItemId).toString());
+        plugin.getConfig().set("notify.any-chest", notifyAnyChest);
+        plugin.getConfig().set("notify.silent-chest", notifySilentChest);
+
+        if (anyChestToggles != null && !anyChestToggles.isEmpty()) {
+            for (Map.Entry<UUID, Boolean> entry : anyChestToggles.entrySet()) {
+                plugin.getConfig().set("toggles.any-chest." + entry.getKey(), entry.getValue());
+            }
+        }
+
+        if (itemOpenInvToggles != null && !itemOpenInvToggles.isEmpty()) {
+            for (Map.Entry<UUID, Boolean> entry : itemOpenInvToggles.entrySet()) {
+                plugin.getConfig().set("toggles.items.open-inv." + entry.getKey(), entry.getValue());
+            }
+        }
+
+        if (silentChestToggles != null && !silentChestToggles.isEmpty()) {
+            for (Map.Entry<UUID, Boolean> entry : silentChestToggles.entrySet()) {
+                plugin.getConfig().set("toggles.silent-chest." + entry.getKey(), entry.getValue());
+            }
+        }
+
+        // Save the new config
+        plugin.saveConfig();
+
+        plugin.getLogger().info("[Config] Update complete.");
+
+    }
+
+    private Map<UUID, Boolean> updateAnyChestToggles() {
```

Comment:
These three methods looks very very similar. We can probably combine these into a single one that takes a parameter of a ConfigurationSection or a String for the key to get from the config.